### ISSUE TITLE
v1.1.36 fails with "The 'class' keyword is not supported in this version of the language"

### ISSUE THREAD

**Issue by roizentner** [BMuuN/vsts-assemblyinfo-task#36]
A build using this task for several months suddenly failed.
The previous build used v1.0.40 and passed.
Here's the console output of the failing build:
`
Preparing task execution handler.
Executing the powershell script: D:\bld2-a1\_tasks\Assembly-Info-Task_3edc451e-360c-40e5-a221-a42d5c987ab8\1.1.36\ApplyAssemblyInfo.ps1
##[error]System.Management.Automation.ParseException: At D:\bld2-a1\_tasks\Assembly-Info-Task_3edc451e-360c-40e5-a221-a42d5c987ab8\1.1.36\Models.ps1:1 char:1
class AssemblyInfo {
The 'class' keyword is not supported in this version of the language.

   at System.Management.Automation.ScriptBlock.Create(Parser parser, String fileName, String fileContents)
   at System.Management.Automation.ExternalScriptInfo.get_ScriptBlock()
   at System.Management.Automation.CommandDiscovery.CreateCommandProcessorForScript(ExternalScriptInfo scriptInfo, ExecutionContext context, Boolean useNewScope, SessionStateInternal sessionState)
   at System.Management.Automation.CommandDiscovery.CreateScriptProcessorForSingleShell(ExternalScriptInfo scriptInfo, ExecutionContext context, Boolean useLocalScope, SessionStateInternal sessionState)
   at System.Management.Automation.CommandDiscovery.LookupCommandProcessor(CommandInfo commandInfo, CommandOrigin commandOrigin, Nullable1 useLocalScope, SessionStateInternal sessionState)
   at System.Management.Automation.CommandDiscovery.LookupCommandProcessor(String commandName, CommandOrigin commandOrigin, Nullable1 useLocalScope)
   at System.Management.Automation.ExecutionContext.CreateCommand(String command, Boolean dotSource)
   at System.Management.Automation.PipelineOps.AddCommand(PipelineProcessor pipe, CommandParameterInternal[] commandElements, CommandBaseAst commandBaseAst, CommandRedirection[] redirections, ExecutionContext context)
   at System.Management.Automation.PipelineOps.InvokePipeline(Object input, Boolean ignoreInput, CommandParameterInternal[][] pipeElements, CommandBaseAst[] pipeElementAsts, CommandRedirection[][] commandRedirections, FunctionContext funcContext)
   at System.Management.Automation.Interpreter.ActionCallInstruction6.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
##[error]System.Management.Automation.PSArgumentException: Cannot find type [AssemblyInfo]: verify that the assembly containing this type is loaded.
   at System.Management.Automation.MshCommandRuntime.ThrowTerminatingError(ErrorRecord errorRecord)
##[error]System.Management.Automation.RuntimeException: The property 'Path' cannot be found on this object. Verify that the property exists and can be set.
   at CallSite.Target(Closure , CallSite , Object , Object )
   at System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)
   at System.Management.Automation.Interpreter.DynamicInstruction3.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
##[error]System.Management.Automation.RuntimeException: The property 'FileNames' cannot be found on this object. Verify that the property exists and can be set.
   at CallSite.Target(Closure , CallSite , Object , Object )
   at System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)
   at System.Management.Automation.Interpreter.DynamicInstruction3.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
##[error]System.Management.Automation.RuntimeException: The property 'Title' cannot be found on this object. Verify that the property exists and can be set.
   at CallSite.Target(Closure , CallSite , Object , Object )
   at System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)
   at System.Management.Automation.Interpreter.DynamicInstruction3.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
##[error]System.Management.Automation.RuntimeException: The property 'Product' cannot be found on this object. Verify that the property exists and can be set.
   at CallSite.Target(Closure , CallSite , Object , Object )
   at System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)
   at System.Management.Automation.Interpreter.DynamicInstruction3.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
##[error]System.Management.Automation.RuntimeException: The property 'Description' cannot be found on this object. Verify that the property exists and can be set.
   at CallSite.Target(Closure , CallSite , Object , Object )
   at System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)
   at System.Management.Automation.Interpreter.DynamicInstruction3.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
##[error]System.Management.Automation.RuntimeException: The property 'Company' cannot be found on this object. Verify that the property exists and can be set.
   at CallSite.Target(Closure , CallSite , Object , Object )
   at System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)
   at System.Management.Automation.Interpreter.DynamicInstruction`3.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
##[error]System.Management.Automation.RuntimeException: The property 'Copyright' cannot be found on this object. Verify that the property exists and can be set.
   at CallSite.Target(Closure , CallSite , Object , Object )
   at System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)
   at System.Management.Automation.Interpreter.DynamicInstruction3.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
##[error]System.Management.Automation.RuntimeException: The property 'Trademark' cannot be found on this object. Verify that the property exists and can be set.
   at CallSite.Target(Closure , CallSite , Object , Object )
   at System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)
   at System.Management.Automation.Interpreter.DynamicInstruction3.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
##[error]System.Management.Automation.RuntimeException: The property 'Culture' cannot be found on this object. Verify that the property exists and can be set.
   at CallSite.Target(Closure , CallSite , Object , Object )
   at System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)
   at System.Management.Automation.Interpreter.DynamicInstruction3.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
##[error]System.Management.Automation.RuntimeException: The property 'Configuration' cannot be found on this object. Verify that the property exists and can be set.
   at CallSite.Target(Closure , CallSite , Object , Object )
   at System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)
   at System.Management.Automation.Interpreter.DynamicInstruction3.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
##[error]System.Management.Automation.RuntimeException: The property 'Version' cannot be found on this object. Verify that the property exists and can be set.
   at CallSite.Target(Closure , CallSite , Object , Object )
   at System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)
   at System.Management.Automation.Interpreter.DynamicInstruction3.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
##[error]System.Management.Automation.RuntimeException: The property 'FileVersion' cannot be found on this object. Verify that the property exists and can be set.
   at CallSite.Target(Closure , CallSite , Object , Object )
   at System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)
   at System.Management.Automation.Interpreter.DynamicInstruction3.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
##[error]System.Management.Automation.RuntimeException: The property 'InformationalVersion' cannot be found on this object. Verify that the property exists and can be set.
   at CallSite.Target(Closure , CallSite , Object , Object )
   at System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)
   at System.Management.Automation.Interpreter.DynamicInstruction3.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
##[error]System.Management.Automation.RuntimeException: The property 'InsertAttributes' cannot be found on this object. Verify that the property exists and can be set.
   at CallSite.Target(Closure , CallSite , Object , Object )
   at System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)
   at System.Management.Automation.Interpreter.DynamicInstruction3.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
##[error]System.Management.Automation.ParameterBindingValidationException: Cannot bind argument to parameter 'Path' because it is null.
   at System.Management.Automation.Internal.PipelineProcessor.SynchronousExecuteEnumerate(Object input, Hashtable errorResults, Boolean enumerate)
   at System.Management.Automation.PipelineOps.InvokePipeline(Object input, Boolean ignoreInput, CommandParameterInternal[][] pipeElements, CommandBaseAst[] pipeElementAsts, CommandRedirection[][] commandRedirections, FunctionContext funcContext)
   at System.Management.Automation.Interpreter.ActionCallInstruction6.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
##[error]System.Management.Automation.RuntimeException: The property 'NetFrameworkFileNames' cannot be found on this object. Verify that the property exists and can be set.
   at CallSite.Target(Closure , CallSite , Object , Object )
   at System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)
   at System.Management.Automation.Interpreter.DynamicInstruction3.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
##[error]System.Management.Automation.RuntimeException: The property 'NetFrameworkFileNames' cannot be found on this object. Verify that the property exists and can be set.
   at CallSite.Target(Closure , CallSite , Object , Object )
   at System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)
   at System.Management.Automation.Interpreter.DynamicInstruction3.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
##[error]System.Management.Automation.RuntimeException: The property 'NetCoreFileNames' cannot be found on this object. Verify that the property exists and can be set.
   at CallSite.Target(Closure , CallSite , Object , Object )
   at System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)
   at System.Management.Automation.Interpreter.DynamicInstruction3.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
##[error]System.Management.Automation.RuntimeException: The property 'Copyright' cannot be found on this object. Verify that the property exists and can be set.
   at CallSite.Target(Closure , CallSite , Object , Object )
   at System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)
   at System.Management.Automation.Interpreter.DynamicInstruction3.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
##[error]System.Management.Automation.RuntimeException: [ref] cannot be applied to a variable that does not exist.
   at System.Management.Automation.VariableOps.GetVariableAsRef(VariablePath variablePath, ExecutionContext executionContext, Type staticType)
   at System.Management.Automation.Interpreter.FuncCallInstruction4.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
Validating Version: 
##[error]System.Management.Automation.RuntimeException: The property 'Version' cannot be found on this object. Verify that the property exists and can be set.
   at CallSite.Target(Closure , CallSite , Object , Object )
   at System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)
   at System.Management.Automation.Interpreter.DynamicInstruction3.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
Validating File Version: 
##[error]System.Management.Automation.RuntimeException: The property 'FileVersion' cannot be found on this object. Verify that the property exists and can be set.
   at CallSite.Target(Closure , CallSite , Object , Object )
   at System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)
   at System.Management.Automation.Interpreter.DynamicInstruction3.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
Validating Informational Version: 
##[error]System.Management.Automation.RuntimeException: The property 'InformationalVersion' cannot be found on this object. Verify that the property exists and can be set.
   at CallSite.Target(Closure , CallSite , Object , Object )
   at System.Dynamic.UpdateDelegates.UpdateAndExecute2[T0,T1,TRet](CallSite site, T0 arg0, T1 arg1)
   at System.Management.Automation.Interpreter.DynamicInstruction3.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
Task Parameters:

	Source Directory: 
	Net Framework File: 
	Net Core File: 
	Title: 
	Product: 
	Description: 
	Company: 
	Copyright: 
	Trademark: 
	Culture: 
	Configuration: 
	Version: 
	File Version: 
	Informational Version: 
	Insert Attributes: 

##[error]PowerShell script completed with 25 errors.
##[section]Finishing: Set Assembly Manifest Data
`

**Comment by BMuuN** [comment]
Apologies @roizentner I'll look into this for you.

Can you confirm if you are using Team Services or a local TFS instance,  if you are using Team Services can you confirm if you are using a hosted build agent or a local private build agent.

**Comment by BMuuN** [comment]
I can't reproduce the issue but I think I know the problem: -
Yesterday I pushed a new version of the extension (v1.1.36) which included a dependency on PowerShell v5.0.  The error you've posted suggests you don't have PowerShell v5.0 so I've removed the dependency from the extension and pushed a new version (v1.1.37).  Please let me know if this version is working for you.

**Comment by roizentner** [comment]
Hi @BMuuN,
To answer your questions: I am using VSTS with a private agent.
Looks like you found the correct reason. I don't have PowerShell v5.0 on the agent machine.
I can confirm v1.1.37 is working for me.
Thanks for the quick fix 👍

**Comment by BMuuN** [comment]
Thanks for getting back to me.
That's epic news I'll close this issue.

**Comment by thijstijsma** [comment]
Posting this in case other people are having the same issue:

We're using a self-hosted instance of VS with on-premise build agents. I'm not sure when it checks for an updated version, but I've tried several builds and it wouldn't update from the broken 1.1.36 to 1.1.38.

When I deleted the 1.1.36 version (folder + .completed file) in "<path to agent>\_work\_tasks\Assembly-Info-Task_3edc451e-360c-40e5-a221-a42d5c987ab8" it downloaded 1.1.38, so it's working fine now.

If anyone has more information on when it checks for updates, I'd appreciate an update.
